server/src-lib/Hasura/GraphQL/Execute/Prepare.hs

```diff
@@ -46,7 +44,7 @@ type PrepArgMap = IntMap.IntMap (Q.PrepArg, PGScalarValue)
 -- | Full execution plan to process one GraphQL query.  Once we work on
 -- heterogeneous execution this will contain a mixture of things to run on the
 -- database and things to run on remote schemas.
-type ExecutionPlan db remote raw = ExecutionStep db remote raw
+type ExecutionPlan db remote raw = InsOrdHashMap Text (ExecutionStep db remote raw)
```

Comment:
Could `ExecutionStep` be made a little less polymorphic to avoid the need for so many type parameters?

1. Could `remote` be hardcoded to `RemoteCall`, or is it sometimes something else?

2. Could `raw` be hardcoded to `Aeson.Value`, or is it sometimes something else?

Reply:
Done. The `db` type variable is used in a rather minor way in `Execute/Query.hs`.